sway-fmt-v2/src/utils/expr/mod.rs

```diff
@@ -0,0 +1,953 @@
+                formatter
+                    .shape
+                    .code_line
+                    .update_expr_kind(ExprKind::Collection);
+
+                // get the length in chars of the code_line in a normal line format
+                let mut buf = FormattedCode::new();
+                let mut temp_formatter = Formatter::default();
+                format_tuple(tuple_descriptor, &mut buf, &mut temp_formatter)?;
+                let body_width = buf.chars().count() as usize;
+
+                formatter.shape.add_width(body_width);
+                formatter
+                    .shape
+                    .get_line_style(None, body_width, &formatter.config);
+
+                format_tuple(tuple_descriptor, formatted_code, formatter)?;
+
+                // revert to previous state
+                formatter.shape.sub_width(body_width);
+                formatter.shape.update_line_settings(prev_state);
+            }
+            Self::Parens(expr) => {
+                Self::open_parenthesis(formatted_code, formatter)?;
+                expr.clone()
+                    .into_inner()
+                    .format(formatted_code, formatter)?;
+                Self::close_parenthesis(formatted_code, formatter)?;
+            }
+            Self::Block(code_block) => {
+                CodeBlockContents::open_curly_brace(formatted_code, formatter)?;
+                code_block
+                    .clone()
+                    .into_inner()
+                    .format(formatted_code, formatter)?;
+                CodeBlockContents::close_curly_brace(formatted_code, formatter)?;
+            }
+            Self::Array(array_descriptor) => {
+                ExprArrayDescriptor::open_square_bracket(formatted_code, formatter)?;
+                array_descriptor
+                    .clone()
+                    .into_inner()
+                    .format(formatted_code, formatter)?;
+                ExprArrayDescriptor::close_square_bracket(formatted_code, formatter)?;
+            }
+            Self::Asm(asm_block) => asm_block.format(formatted_code, formatter)?,
+            Self::Return {
+                return_token,
+                expr_opt,
+            } => {
+                write!(formatted_code, "{}", return_token.span().as_str())?;
+                if let Some(expr) = &expr_opt {
+                    write!(formatted_code, " ")?;
+                    expr.format(formatted_code, formatter)?;
+                }
+            }
+            Self::If(if_expr) => if_expr.format(formatted_code, formatter)?,
+            Self::Match {
+                match_token,
+                value,
+                branches,
+            } => {
+                write!(formatted_code, "{} ", match_token.span().as_str())?;
+                value.format(formatted_code, formatter)?;
+                MatchBranch::open_curly_brace(formatted_code, formatter)?;
+                let branches = branches.clone().into_inner();
```

Comment:
I've noticed there are quite a lot of places where we `clone()` in order to call `into_inner()`, even though we only need the inner type by reference.

In this case, the `Braces` type has [a `get` method](https://docs.rs/sway-ast/0.19.2/sway_ast/brackets/struct.Braces.html#method.get) which allows us to use the existing instance directly rather than cloning the whole structure.

It might be worth checking all our uses of `.clone().into_inner()` for similar cases to this as we can likely avoid quite a bit of unnecessary allocation.